apis/v1alpha1/shared_types.go

```diff
@@ -175,6 +189,31 @@ type AdminNetworkPolicyEgressPeer struct {
 	// <network-policy-api:experimental>
 	// +optional
 	Nodes *metav1.LabelSelector `json:"nodes,omitempty"`
+	// Networks defines a way to select peers via CIDR blocks.
+	// This is intended for representing entities that live outside the cluster,
+	// which can't be selected by pods, namespaces and nodes peers, but note
+	// that cluster-internal traffic will be checked against the rule as
+	// well. So if you Allow or Deny traffic to `"0.0.0.0/0"`, that will allow
+	// or deny all IPv4 pod-to-pod traffic as well. If you don't want that,
+	// add a rule that Passes all pod traffic before the Networks rule.
+	//
+	// Note that because policies are applied after Service VIPs (clusterIPs, externalIPs,
+	// load balancer IPs) are rewritten to endpoint IPs, a Networks selector cannot match
+	// such a VIP. For example, a Networks selector that denies traffic to the entire
+	// service CIDR will not actually block any service traffic.
```

Comment:
What if the VIP has no backends/endpoints? The implementation can't translate the traffic, so at that point is the traffic subject to CIDR policy or not? I believe that the behaviour in Cilium today is to drop this traffic, but not motivated by policy at all (before the policy check).

(Other than that, the "translate first, apply policy after" pattern is how Cilium works with respect to policy applying to service traffic.)

Reply:
Ah! I think this question is also related to the `LoadBalancer` behaviour question above :sweat_smile: https://github.com/kubernetes-sigs/network-policy-api/pull/185#discussion_r1506114425

Reply:
yes right there is a discrepancy with LBs will open a new issue to track that.